diff --git a/flexc++/scanner/scanner.h b/flexc++/scanner/scanner.h
index 6661901..18c9d59 100644
--- a/flexc++/scanner/scanner.h
+++ b/flexc++/scanner/scanner.h
@@ -86,8 +86,12 @@ class Scanner: public ScannerBase, private FlexTypes
         using ScannerBase::push;
 
         void push(StartCondition__ sc); // push and revert to sc
-        int popSc(int token = 0);   // revert to the pushed StartCondition,
-                                    // returning 'token'
+        int popSc(int token);   // revert to the pushed StartCondition,
+                                // returning 'token'
+        inline int popSc()
+        {
+            return popSc(0);
+        }
 
             // the start-condition stack can only be used from push and popSC
         class SCStack: private std::stack<StartCondition__> 
